logic/tool/TypeResolver.java

```diff
@@ -220,7 +253,7 @@ private void registerHas(TypeVariable owner, HasConstraint hasConstraint) {
         private void registerRelation(TypeVariable owner, RelationConstraint constraint) {
             for (RelationConstraint.RolePlayer rolePlayer : constraint.players()) {
                 TypeVariable playerResolver = register(rolePlayer.player());
-                TypeVariable actingRoleResolver = register(new TypeVariable(newSystemId()));
+                TypeVariable actingRoleResolver = new TypeVariable(newSystemId());
```

Comment:
I'm not quite following - do we do this for all new system-id generated TypeVariables? How does this not break the fact that we do want to put resolved labels on anonymous variables in when doing normal type resolution?

Reply:
Because this is not an anonymous variable. When a `roleType` is missing, it simply is missing, there is no anonymous variable to add `resolvedTypes` to.